ompi/mca/mtl/ofi/mtl_ofi.h

```diff
@@ -247,13 +247,20 @@ ompi_mtl_ofi_send_start(struct mca_mtl_base_module_t *mtl,
     endpoint = ompi_mtl_ofi_get_endpoint(mtl, ompi_proc);
 
     ompi_ret = ompi_mtl_datatype_pack(convertor, &start, &length, &free_after);
-    if (OMPI_SUCCESS != ompi_ret) return ompi_ret;
+	if (OPAL_UNLIKELY(OMPI_SUCCESS != ompi_ret)) {
+		return ompi_ret;
+	}
 
     ofi_req->buffer = (free_after) ? start : NULL;
     ofi_req->length = length;
     ofi_req->status.MPI_ERROR = OMPI_SUCCESS;
 
-    if (OPAL_UNLIKELY(MCA_PML_BASE_SEND_SYNCHRONOUS == mode)) {
+	if (OPAL_UNLIKELY(length > endpoint->mtl_ofi_module->max_msg_size)) {
```

Comment:
Similar to libfabric, this check seems unnecessary (because it should be obeyed by higher layers).

Do you want to enable this check only in debug mode, perchance?

Reply:
It's an MTL so there are no higher layers to handle the error case. The next level here is directly the MPI user, and as a result any error returned to the higher level will automatically translate into an MPI error and thus an abort.

Reply:
`cm` doesn't handle this case?

Reply:
If `cm` doesn't handle this, why does setting `ompi_mtl_ofi.max_msg_size` matter?

Reply:
max_inject_size is also referenced in ompi_mtl_ofi_send().

Reply:
I totally missed that this is an OFI-module-specific value; it's not on a struct that `cm` groks.  Got it.

Reply:
I am aware, however, that this is a performance sensitive path - it's the reason I used OPAL_UNLIKELY(). Speaking of which, I also wrapped an OPAL_UNLIKELY around the check for a failure from ompi_mtl_datatype_pack() - because that's only going to return an error if malloc() has failed.